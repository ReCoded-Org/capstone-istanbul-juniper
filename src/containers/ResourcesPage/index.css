.resourcesPage {
  padding: 30px;
  background-color: #b3d0c4;
}

.resourcesPage__headerContainer {
  color: black;
}

.resourcesPage__headerContainer__title {
  font-family: "Salsa";
  font-size: 30px;
}

.resourcesPage__headerContainer__subTitle {
  max-width: 400px;
  width: 100%;
}

.resourcesPage__resourcesContainer {
  margin-top: 50px;
  display: grid;
  gap: 30px;
}
