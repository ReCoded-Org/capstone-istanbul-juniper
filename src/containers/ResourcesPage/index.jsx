import React, { useState } from "react";
import Resource from "../../components/Resource";
import "./index.css";
import ourClimateChange from "../../images/ourClimateChange.png";
import resourcesForEducation from "../../images/resourcesForEducation.png";
import whoCausesClimate from "../../images/whoCausesClimate.png";
import { useTranslation } from "react-i18next";

const ResourcesPage = () => {
  const [t] = useTranslation();
  const initResources = [
    {
      id: 1,
      title: t("resourcesPage.resources.climateChangeHandbook.title"),
      body: t("resourcesPage.resources.climateChangeHandbook.text"),
      pic: ourClimateChange,
      url:
        "gs://capstone-istanbul-juniper.appspot.com/ResourcesPage/Child friendly climate change handbook.pdf",
    },
    {
      id: 2,
      title: t(
        "resourcesPage.resources.educationOnClimateChangeHandbook.title"
      ),
      body: t("resourcesPage.resources.educationOnClimateChangeHandbook.text"),
      pic: resourcesForEducation,
      url:
        "gs://capstone-istanbul-juniper.appspot.com/ResourcesPage/Resources list_Climate Change.pdf",
    },
    {
      id: 3,
      title: t("resourcesPage.resources.whoCausesClimateChangesHandbook.title"),
      body: t("resourcesPage.resources.whoCausesClimateChangesHandbook.text"),
      pic: whoCausesClimate,
      url:
        "gs://capstone-istanbul-juniper.appspot.com/ResourcesPage/creating-futures-lesson-4.pdf",
    },
  ];
  const [resources] = useState(initResources);
  return (
    <div className="resourcesPage">
      <div className="resourcesPage__headerContainer">
        <div className="resourcesPage__headerContainer__title">
          {t("resourcesPage.title")}
        </div>
        <div className="resourcesPage__headerContainer__subTitle">
          {t("resourcesPage.aboutPage")}
        </div>
      </div>
      <div className="resourcesPage__resourcesContainer">
        {resources.map((resource) => {
          return (
            <Resource key={`resource_${resource.id}`} resource={resource} />
          );
        })}
      </div>
    </div>
  );
};

export default ResourcesPage;
