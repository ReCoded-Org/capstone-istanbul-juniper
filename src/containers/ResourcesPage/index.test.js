import React from "react";
import renderer from "react-test-renderer";
import ResourcesPage from ".";

it("ResourcesPage renders correctly", () => {
  const tree = renderer.create(<ResourcesPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
