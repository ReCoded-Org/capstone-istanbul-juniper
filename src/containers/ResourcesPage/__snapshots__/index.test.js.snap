// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourcesPage renders correctly 1`] = `
<div
  className="resourcesPage"
>
  <div
    className="resourcesPage__headerContainer"
  >
    <div
      className="resourcesPage__headerContainer__title"
    >
      resourcesPage.title
    </div>
    <div
      className="resourcesPage__headerContainer__subTitle"
    >
      resourcesPage.aboutPage
    </div>
  </div>
  <div
    className="resourcesPage__resourcesContainer"
  >
    <div
      className="resource__body"
    >
      <div
        className="resource__body__picContainer"
      >
        <img
          alt="resourcesPage.resources.climateChangeHandbook.title"
          onClick={[Function]}
          src="ourClimateChange.png"
        />
      </div>
      <div
        className="resource__body__content"
      >
        <div
          className="resource__body__content__title"
        >
          resourcesPage.resources.climateChangeHandbook.title
        </div>
        <div
          className="resource__body__content__description"
        >
          resourcesPage.resources.climateChangeHandbook.text
        </div>
        <div
          className="resource__body__content__footer"
        >
          <button
            onClick={[Function]}
          >
            Visit Resource
          </button>
        </div>
      </div>
    </div>
    <div
      className="resource__body"
    >
      <div
        className="resource__body__picContainer"
      >
        <img
          alt="resourcesPage.resources.educationOnClimateChangeHandbook.title"
          onClick={[Function]}
          src="resourcesForEducation.png"
        />
      </div>
      <div
        className="resource__body__content"
      >
        <div
          className="resource__body__content__title"
        >
          resourcesPage.resources.educationOnClimateChangeHandbook.title
        </div>
        <div
          className="resource__body__content__description"
        >
          resourcesPage.resources.educationOnClimateChangeHandbook.text
        </div>
        <div
          className="resource__body__content__footer"
        >
          <button
            onClick={[Function]}
          >
            Visit Resource
          </button>
        </div>
      </div>
    </div>
    <div
      className="resource__body"
    >
      <div
        className="resource__body__picContainer"
      >
        <img
          alt="resourcesPage.resources.whoCausesClimateChangesHandbook.title"
          onClick={[Function]}
          src="whoCausesClimate.png"
        />
      </div>
      <div
        className="resource__body__content"
      >
        <div
          className="resource__body__content__title"
        >
          resourcesPage.resources.whoCausesClimateChangesHandbook.title
        </div>
        <div
          className="resource__body__content__description"
        >
          resourcesPage.resources.whoCausesClimateChangesHandbook.text
        </div>
        <div
          className="resource__body__content__footer"
        >
          <button
            onClick={[Function]}
          >
            Visit Resource
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
